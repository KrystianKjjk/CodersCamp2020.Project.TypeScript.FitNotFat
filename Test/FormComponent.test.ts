import generateForm from '../Src/UIComponents/FormComponent/FormComponent';

describe(('tests for Form Component'),()=>{

    test('Should render correctly',()=>{
       const form=generateForm();
       expect(form).toMatchSnapshot();
    })

    test("should change type to date on focus", () => {
        const form = generateForm();
        document.body.appendChild(form);
        const birthDateInput = document.body.querySelector('.birth-date') as HTMLInputElement;
        expect(birthDateInput.type).toBe("text");
        birthDateInput.focus();
        expect(birthDateInput.type).toBe("date");
    })
})