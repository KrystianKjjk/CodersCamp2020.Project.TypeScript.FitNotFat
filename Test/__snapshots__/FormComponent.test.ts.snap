// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for Form Component Should render correctly 1`] = `
<form>
  <p>
    Join us
  </p>
  <div
    class="progress-bar"
  >
    <div
      class="first-step"
    />
  </div>
  <input
    class="name"
    minlength="3"
    placeholder="Nickname"
    required=""
    type="text"
  />
  <div
    class="gender-selection"
  >
    <input
      id="male"
      name="gender"
      required=""
      type="radio"
      value="male"
    />
    <label
      for="male"
    >
      Male
    </label>
    <span />
    <input
      class="radio-button"
      id="female"
      name="gender"
      required=""
      type="radio"
      value="female"
    />
    <label
      for="female"
    >
      Female
    </label>
  </div>
  <input
    class="birth-date"
    placeholder="Date of birth"
    required=""
    type="text"
  />
  <div
    class="current-values"
  >
    <input
      class="height"
      min="0"
      placeholder="Height (cm)"
      required=""
      type="number"
    />
    <input
      class="current-weight"
      min="0"
      placeholder="Current weight (kg)"
      required=""
      type="number"
    />
  </div>
  <input
    class="goal-weight"
    min="0"
    placeholder="Goal weight (kg)"
    required=""
    type="number"
  />
  <button
    class="button"
    style="background-color: rgb(218, 27, 54); color: white; height: 50px; width: 35%;"
  />
</form>
`;
